web/src/message_view_header.ts

```diff
@@ -70,6 +70,18 @@ function get_message_view_header_context(filter: Filter | undefined): MessageVie
     //
     // TODO: This ideally doesn't need a special case, we can just use
     // `filter.get_description` for it.
+
+    if (filter?.has_negated_operand("is", "dm")) {
```

Comment:
Why are you checking just a single operand? I think the description here seems only correct if this is the only operator.